crates/biome_js_analyze/src/semantic_analyzers/nursery/no_misleading_character_class.rs

```diff
@@ -0,0 +1,627 @@
+use crate::{semantic_services::Semantic, JsRuleAction};
+use biome_analyze::{context::RuleContext, declare_rule, ActionCategory, Rule, RuleDiagnostic};
+use biome_console::markup;
+use biome_diagnostics::Applicability;
+use biome_js_factory::make;
+use biome_js_syntax::{
+    AnyJsCallArgument, AnyJsExpression, AnyJsLiteralExpression, AnyJsTemplateElement,
+    JsCallArguments, JsCallExpression, JsNewExpression, JsRegexLiteralExpression,
+    JsStringLiteralExpression, JsSyntaxKind, JsSyntaxToken, T,
+};
+use biome_rowan::{
+    declare_node_union, AstNode, AstNodeList, AstSeparatedList, BatchMutationExt, TextRange,
+};
+declare_rule! {
+    /// Disallow characters which are made with multiple code points in character class syntax
+    ///
+    /// Unicode includes the characters which are made with multiple code points. e.g. Á, 🇯🇵, 👨‍👩‍👦.
+    /// RegExp character class syntax (/[abc]/) cannot handle characters which are made by multiple code points as
```

Comment:
We could add an example:

```suggestion
    /// A RegExp character class `/[abc]/` cannot handle characters that consits of multiple code points.
    /// For example, the character `❇️` consists of two code points: `❇` (U+2747) and `VARIATION SELECTOR-16` (U+FE0F).
    /// If this character is in a RegExp character class, it will match to either `❇` or `VARIATION SELECTOR-16` rather than `❇️`.
```